.github/workflows/build_publish.yaml

```diff
@@ -5,13 +5,26 @@ on: [push]
 jobs:
   build:
     name: build and publish voltcraft
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+
     steps:
     - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v4
       with:
-        python-version: 3.11
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install pre-commit
+    - name: Lint, mypy and formatting
+      run: |
+        pre-commit run -a
```

Comment:
I would suggest to remove this for now, and I will add pre-commit.ci after we merge your PR, so that the pre-commit config is enforced that way.

Reply:
Back to origin github actions